### PR TITLE
refactor: split and optimize config type

### PR DESCRIPTION
**Pull request by Aya0wind**
split config type to session, proxy, jodis, backend and metrics.And change protocol_type to enum.

### CODE REVIEW

File: src/proxy/proxy.rs [hunk 1]
```diff
@@ -30,18 +30,18 @@ struct RawProxy {
...
         let proxy = RawProxy {
             xauth: String::new(),
 
             online: false,
             closed: false,
 
-            lproxy_addr: SocketAddr::from_str(&config.proxy_addr()).unwrap(),
-            ladmin_addr: SocketAddr::from_str(&config.admin_addr()).unwrap(),
+            lproxy_addr: SocketAddr::from_str(&config.proxy.proxy_addr).unwrap(),
+            ladmin_addr: SocketAddr::from_str(&config.proxy.proxy_addr).unwrap(),
```
Comment (yaoyinnan) [inline src/proxy/proxy.rs:44]: typo? Should it be `admin_addr`?

File: src/proxy/config.rs [hunk 2]
```diff
@@ -1,85 +1,134 @@
...
+    pub proxy_max_offheap_bytes: u64,
     #[serde(deserialize_with = "deserialize_string_to_size")]
-    session_send_bufsize: u64,
+    pub proxy_heap_place_holder: u64,
+}
+
+/// configuration for metrics
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(default)]
+pub struct MetricsConfig {
```
Comment (yaoyinnan) [inline src/proxy/config.rs:81]: In order to facilitate expansion, would it be better to split `MetricsConfig` into `MetricsJson`, `MetricsInfluxdb`, `MetricsStats`?

File: src/proxy/config.rs [hunk 3]
```diff
@@ -1,85 +1,134 @@
...
+use std::{fs::File, io::Read, path::Path, time::Duration};
 
 const KB: u64 = 1024;
 const MB: u64 = 1024 * KB;
 const GB: u64 = 1024 * MB;
 const TB: u64 = 1024 * GB;
 const PB: u64 = 1024 * TB;
 
-// 定义 Proxy 启动时候的参数列表
+///configuration for jodis
```
Comment (yaoyinnan) [inline src/proxy/config.rs:11]: missing a space

**Review by yaoyinnan — COMMENTED** [review]
little comments

**Review by Dennis-Zhang-SH — APPROVED** [review]

**Review by yaoyinnan — APPROVED** [review]
LGTM

File: src/proxy/config.rs (hunk 2 above)
Comment (infdahai) [reply src/proxy/config.rs:76]: > In order to facilitate expansion, would it be better to split `MetricsConfig` into `MetricsJson`, `MetricsInfluxdb`, `MetricsStats`?

It better be that way.

**Pull request merged**